src/client/src/auth/RautGuard.tsx

```diff
@@ -0,0 +1,23 @@
+import { Outlet, useNavigate } from "@solidjs/router";
+import { createEffect } from "solid-js";
+import Header from "../components/Header";
+import clerk from "./Auth";
+
+const RouteGuard = () => {
```

Comment:
q: why is the component called `RouteGuard` but the file is called `RautGuard`?

Reply:
ups, typing is hard

Reply:
nice catch!